api/src/main/java/jakarta/data/metamodel/expression/FunctionExpression.java

```diff
@@ -15,13 +15,13 @@
  *
  * SPDX-License-Identifier: Apache-2.0
  */
-package jakarta.data.metamodel.restrict;
+package jakarta.data.metamodel.expression;
 
-import jakarta.data.metamodel.constraint.Constraint;
+import jakarta.data.metamodel.Expression;
 
-public interface TextRestriction<T> extends BasicRestriction<T, String> {
-    @Override
-    TextRestriction<T> negate();
-    @Override
-    Constraint<String> constraint();
+import java.util.List;
+
+public interface FunctionExpression<T,V> extends Expression<T,V> {
+    String name();
```

Comment:
We can fix this later, but it seems like this should be an enumeration of possible functions. Or if not that, at least there should be String constants for each function name defined by the spec.

Reply:
I think string constants would be a good idea. I'm working on the assumption that functions are eventually going to be an open world.

Reply:
Done!